File: 3 занятие/I.py
def isPrime(n):
    if n <= 1:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

n = int(input())

if n > 2 and n % 2 == 0:
    for i in range(1, n):
        if isPrime(i) and isPrime(n - i):
            print(i, n - i)
            break
else:
    if isPrime(n - 2):
        print(2, n - 2)
    else:
        print(-1)

File: 3 занятие/I.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

bool isPrime(ll n) {
    if (n <= 1) return false;
    for (int i = 2; i * i <= n; i++)
        if (n % i == 0) 
            return false;
    return true;
}

int main() {
    ll n;
    cin >> n;

    if (n > 2 && n % 2 == 0) {
        for (int i = 1; i < n; i++) {
            if (isPrime(i) && isPrime(n - i)) {
                cout << i << " " << n - i;
                break;
            }
        }
    } else {
        if (isPrime(n - 2)) {
            cout << 2 << " " << n - 2;
        } else {
            cout << -1;
        }
    }

}